utility/utility.go

```diff
@@ -72,7 +74,23 @@ func AllZero(s []byte) bool {
 }
 
 func SanitizePath(path string) string {
-	return strings.TrimLeft(path, "/")
+	return strings.TrimLeft(path, PathSeparator)
+}
+
+func NormalizePath(path string) string {
+	return strings.TrimRight(path, PathSeparator)
+}
+
+func IsInDirectory(path, directoryPath string) bool {
+	relativePath, err := filepath.Rel(directoryPath, path)
+	if err != nil {
+		return false
+	}
+	return relativePath == "." || NormalizePath(NormalizePath(directoryPath)+PathSeparator+relativePath) == NormalizePath(path)
```

Comment:
Join is too smart.
path.Join("a/..", "b") returns "b".
So if i use join in here, this function would check nothing.

Reply:
I think, just checking if one path starts with another would work.

Reply:
Didn't work. Returned it back. I will think about it tomorrow. I guess, the rest should be ok.

Reply:
Checking if one path is a prefix on another is not a good idea, because it would work for any two strings even if they are not valid paths. filepath.Rel would return error for such case. So i think it's best to leave current version.